common/src/main/java/com/wynntils/core/events/EventThread.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.events;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Inform what type of thread the event is allowed to be sent on. Events without an
+ * explicit annotation is considered to be allowed on Type == RENDER only.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface EventThread {
+    Type value() default Type.RENDER;
```

Comment:
I might've even removed the default value, such that people get used to defining it.

Reply:
When I started the annotation process, it turned out that almost all events are thrown (correctly) on the main/render thread. So this annotation is really about the few events that are not. If anything, I could remove RENDER, and force you to declare using the Render thread by not having the annotation at all, but I thought I would keep it to make it possible to tell this explicitly.